lib/big_brother/node.rb

```diff
@@ -30,7 +30,7 @@ def incorporate_state(another_node)
     end
 
     def invalidate_weight!
-      @weight = nil
+      @weight = 0
```

Comment:
`nil` can cause issues in the code path, especially in `#monitor_nodes` where there is a check (`node.weight.zero?`) on node weight. If the monitor runs as the cluster is being shut down, this can error out. We've seen this in cases of particularly heavy load.

You make a good point though, and there is one edge case where nil is helpful and we need to handle this change. In the same `#monitor_nodes` method, we check the current weight and the measured weight. If a new cluster's node weights are now set to 0 and the measured health of the node is 0 (because it is down or intentionally set to 0), then the measured and node weights are the same and the piece to update the node is not hit, leaving bigbro and ipvs diverged. [28c48d4](https://github.com/braintree/big_brother/pull/17/commits/28c48d4bd05a0c9f3ae0e9e76ed02687a6e32b40) handles this case.